instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/QueryParameterKeyCache.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.servlet;
+
+import io.opentelemetry.api.common.AttributeKey;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+class QueryParameterKeyCache {
+
+  public static final String KEY_PREFIX = "servlet.request.parameter.";
+  private static final ConcurrentMap<String, AttributeKey<List<String>>> cache =
```

Comment:
perhaps `Cache.bounded()` would be suitable here

Reply:
Nice, didn't realize we had that! 👍🏻